src/MonitorProfiler/ProbeInstrumentation/ProbeInstrumentation.cpp

```diff
@@ -299,7 +299,15 @@ bool ProbeInstrumentation::AreProbesInstalled()
 
 void ProbeInstrumentation::AddProfilerEventMask(DWORD& eventsLow)
 {
-    eventsLow |= COR_PRF_MONITOR::COR_PRF_ENABLE_REJIT;
+    //
+    // Workaround:
+    // Enable COR_PRF_MONITOR_JIT_COMPILATION even though we don't need the callbacks.
+    // It appears that without this flag set our RequestReJITWithInliners calls will sometimes
+    // not actually trigger a rejit despite returning successfully.
+    //
+    // This issue most commonly occurs on MacOS.
```

Comment:
Has it been "most commonly occurs on MacOS" or "always and only occurs on MacOS"? I would prefer to scope it to the platform that has the issue if it is platform dependent rather than making all variants pay the cost for JIT compilation callbacks.